jujugui/static/gui/src/app/app.js

```diff
@@ -1011,11 +1011,82 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Get user name from cookie.
+
+      @method _getUsernameFromCookie
+    */
+    _getUsernameFromCookie: function() {
+      // XXX jcsackett 2016-01-08: We should be using the whoami endpoint on
+      // charmstore to get the username of a logged in jem user rather than
+      // parsing the cookie. Consider this all a bad hack for demo and we'll do
+      // something better in the very near future.
+
+      // Don't look for cookies if in sandbox.
+      if (this.get('sandbox')) {
+        return null;
+      }
+      var cookie = document.cookie;
+      var data = cookie.split(';');
+      var macaroon = null;
+      var prefix = 'Macaroons-jem=';
+      var found = data.some(function(item) {
+        macaroon = item;
+        return (item.indexOf(prefix) !== -1);
+      });
+      if (!found) {
+        console.error('No macaroon found to get username from.');
+        return null;
+      }
+      var index = macaroon.indexOf(prefix) + prefix.length;
+      macaroon.slice(index);
+      macaroon = macaroon.replace('Macaroons-jem=', '');
+      macaroon = JSON.parse(atob(macaroon));
```

Comment:
That's presumably also true even if it's valid JSON. Say, for example it was a JSON string not an object,
then I'm guessing that the forEach loop below would fall over. It might be better to use the macaroons library to parse it, which should do appropriate verification. Or just put a try/catch around the whole function.